Docs: Add documentation for next/third-parties (#54805)
vercel/next.js: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx

---

> title: Third Parties

Reviewer: @janicklas-ralph Can you update the examples (or create new ones) in the Next.js examples repo in this PR for each of the newer third-parties included?

We should have `with-google-maps` and `with-youtube-embed` updated (or created if they don't yet exist) that shows how to use these newer components.

nav_title: Third Parties
description: Optimize the performance of third-party libraries in your application with the `@next/third-parties` package.
---

{/* The content of this doc is shared between the app and pages router. You can use the `<PagesOnly>Content</PagesOnly>` component to add content that is specific to the Pages Router. Any shared content should not be wrapped in a component. */}

> [**`@next/third-parties`**](/docs/app/api-reference/components/font) is a library that provides a

Reviewer: ^ We should remove the link if we're not including an API reference page

| Name              | Type     | Description                                                                                         |
| ----------------- | -------- | --------------------------------------------------------------------------------------------------- |
| `apiKey`          | Required | Your api key.                                                                                       |
| `mode`            | Required | [Map mode](https://developers.google.com/maps/documentation/embed/embedding-map#choosing_map_modes) |
| `height`          | Optional | Height of the embed. Defaults to `auto`.                                                            |
| `width`           | Optional | Width of the embed. Defaults to `auto`.                                                             |
| `style`           | Optional | Pass styles to the iframe.                                                                          |
| `allowfullscreen` | Optional | Property to allow certain map parts to go full screen.                                              |

> | `loading`         | Optional | Defaults to lazy.                                                                                   |

Reviewer 2: Maybe it makes sense to clarify when we recommend changing the value.

```suggestion
| `loading`         | Optional | Defaults to lazy. Consider changing if you know your embed will be above-the-fold.                                                                                 |
```

> | `q`               | Optional | Defines map marker location. _This maybe required depending on the map mode_.                       |

Reviewer: ```suggestion
| `q`               | Optional | Defines map marker location. _This may be required depending on the map mode_.                       |
```

}
```

</PagesOnly>

#### Options

| Name        | Type     | Description                                                                                                                                                                                                |
| ----------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `videoid`   | Required | YouTube video id.                                                                                                                                                                                          |
| `width`     | Optional | Width of the video container. Defaults to `auto`                                                                                                                                                           |
| `height`    | Optional | Height of the video container. Defaults to `auto`                                                                                                                                                          |

> | `playlabel` | Optional | A visually hidden label for the play button for accessability.                                                                                                                                             |

Reviewer 2: ```suggestion
| `playlabel` | Optional | A visually hidden label for the play button for accessibility.                                                                                                                                             |
```
